src/Umbraco.Web/Models/Mapping/MemberModelMapper.cs

```diff
@@ -477,8 +481,41 @@ public MemberDisplay Convert(ResolutionContext context)
                 return AutoMapperExtensions.MapWithUmbracoContext<IMember, MemberDisplay>(member, context.GetUmbracoContext());
             }
         }
-    }
-}
 
+        internal class MemberSensitivePropertiesResolver : SensitivePropertiesResolver<IMember>
+        {
 
+            /// <summary>
+            /// Overridden to assign the IsSensitive property values
+            /// </summary>
+            /// <param name="umbracoContext"></param>
+            /// <param name="content"></param>
+            /// <param name="properties"></param>
+            /// <returns></returns>
+            protected override List<ContentPropertyDisplay> MapProperties(UmbracoContext umbracoContext, IContentBase content, List<Property> properties)
+            {
+                var result = base.MapProperties(umbracoContext, content, properties);
+                var member = (IMember)content;
+                var memberType = member.ContentType;
 
+                //now update the IsSensitive value
+                foreach (var prop in result)
+                {
+                    //check if this property is flagged as sensitive
+                    var isSensitiveProperty = memberType.IsSensitiveProperty(prop.Alias);
+                    //check permissions for viewing sensitive data
+                    if (isSensitiveProperty && umbracoContext.Security.CurrentUser.HasAccessToSensitiveData() == false)
+                    {
+                        //mark this property as readonly so that it does not post any data
+                        prop.Readonly = true;

```

Comment:
Hrm, this one is a bit odd and this is sort of working by fluke. My previous comment about the `ContentPropertyDisplay` vs `ContentPropertyBasic` is why this is working. This is constructing a `ContentPropertyDisplay` instead of a basic which contains these properties and although the destination mapping is only expecting a "basic", we are giving it a "display" and by the nature of how newtonsoft.json serializes things, we end up serializing the "display" properties to the client even though those properties are not part of the type its supposed to be returned.

Instead we'll need to update the `ContentPropertyBasic` model to carry some information of if the property is sensitive.